tests/Functional/SQL/ParserTest.php

```diff
@@ -17,4 +18,31 @@ public function testPostgreSQLJSONBQuestionOperator(): void
 
         self::assertTrue($this->connection->fetchOne('SELECT \'{"a":null}\'::jsonb ?? :key', ['key' => 'a']));
     }
+
+    public function testPostgresPlaceholders(): void
+    {
+        if (! TestUtil::isDriverOneOf('pgsql')) {
+            self::markTestSkipped('This test requires the pgsql driver.');
+        }
+
+        $this->connection->beginTransaction();
+        $this->connection->executeStatement(
+            'CREATE TABLE dummy_table ' .
+            '(a_number smallint, a_number_2 smallint, b_number smallint, c_number smallint, a_number_3 smallint)',
+        );
+        $this->connection->executeStatement(
+            'INSERT INTO dummy_table (a_number, a_number_2, b_number, c_number, a_number_3)' .
+            ' VALUES ($1, $1, $2, $3, $1)',
+            [1, 2, 3],
+            [ParameterType::INTEGER, ParameterType::INTEGER, ParameterType::INTEGER],
+        );
+        $result = $this->connection->executeQuery('SELECT * FROM dummy_table')->fetchAllAssociative();
+        self::assertCount(1, $result);
+        self::assertEquals(1, $result[0]['a_number']);
+        self::assertEquals(1, $result[0]['a_number_2']);
+        self::assertEquals(2, $result[0]['b_number']);
+        self::assertEquals(3, $result[0]['c_number']);
+        self::assertEquals(1, $result[0]['a_number_3']);
+        $this->connection->rollBack();
```

Comment:
This probably needs to be done in the `finally` block. Otherwise, if the test fails, the transaction will remain uncommitted.

Even better, just remove the transaction and use `dropTableIfExists()` before creating it.

Reply:
Good catch, rewrote it using `dropAndCreateTable` instead